autorag/utils/util.py

```diff
@@ -17,14 +17,27 @@
 
 def fetch_contents(corpus_data: pd.DataFrame, ids: List[List[str]],
                    column_name: str = 'contents') -> List[List[Any]]:
-    flat_ids = itertools.chain.from_iterable(ids)
-    contents = list(map(lambda x: corpus_data.loc[lambda row: row['doc_id'] == x][column_name].values[0], flat_ids))
+    flat_ids = [item for sublist in ids for item in (sublist if sublist else [None])]
```

Comment:
Just use `itertools.chain.from_iterable(ids)` for flatten ids.
The input is just 2-d list.

Reply:
But, I have to put [None] if ids has empty list. 
`itertools.chain.from_iterable(ids)` We can't put [None] with this code alone...